src/pages/Account/Error.tsx

```diff
@@ -8,22 +8,30 @@ type ErrorProps = {
 };
 
 export default function Error({error, address}: ErrorProps) {
-  if (error.type == ResponseErrorType.NOT_FOUND) {
-    return (
-      <Alert severity="error" sx={{overflowWrap: "break-word"}}>
-        {error.message}
-        Account not found: {address}
-      </Alert>
-    );
-  } else {
-    return (
-      <Alert severity="error">
-        Unknown error ({error.type}) fetching an Account with address {address}:
-        <br />
-        {error.message}
-        <br />
-        Try again later
-      </Alert>
-    );
+  switch (error.type) {
+    case ResponseErrorType.NOT_FOUND:
+      return (
+        <Alert severity="error" sx={{overflowWrap: "break-word"}}>
+          {error.message}
+          Account not found: {address}
+        </Alert>
+      );
+    case ResponseErrorType.UNHANDLED:
+      return (
+        <Alert severity="error">
+          Unknown error ({error.type}) fetching an Account with address{" "}
+          {address}:
+          <br />
+          {error.message}
+          <br />
+          Try again later
+        </Alert>
+      );
+    case ResponseErrorType.TOO_MANY_REQUESTS:
+      return (
+        <Alert severity="error">
+          To many requests. Please try again 5 minutes later.
+        </Alert>
+      );
   }
```

Comment:
do we need default case here? or it always falls under UNHANDLED if not 429 and not found

Reply:
yeah UNHANDLED is the default experience rn. we don't easily use <Error> component only if thats needed, so UNHANDLED won't get triggered easily.